example/gateway.js

```diff
@@ -1,13 +1,28 @@
 const { ApolloServer } = require('apollo-server');
 const { ApolloGateway } = require('@apollo/gateway');
+const {
+  RemoteGraphQLDataSource,
+} = require('@apollo/gateway/dist/datasources/RemoteGraphQLDataSource');
+const GrpcGraphQLDataSource = require('./GrpcGraphQLDataSource').default;
 
 const gateway = new ApolloGateway({
   serviceList: [
-    { name: 'accounts', url: 'http://localhost:5001/graphql' },
-    { name: 'reviews', url: 'http://localhost:5002/graphql' },
-    { name: 'products', url: 'http://localhost:5003/graphql' },
-    { name: 'inventory', url: 'http://localhost:5004/graphql' },
+    // { name: 'accounts', url: 'http://localhost:50001/graphql' },
+    // { name: 'reviews', url: 'http://localhost:50002/graphql' },
+    // { name: 'products', url: 'http://localhost:50003/graphql' },
+    // { name: 'inventory', url: 'http://localhost:50004/graphql' },
+
+    { name: 'accounts', url: 'localhost:50001', type: 'grpc' },
+    { name: 'reviews', url: 'localhost:50002', type: 'grpc' },
+    { name: 'products', url: 'localhost:50003', type: 'grpc' },
+    { name: 'inventory', url: 'localhost:50004', type: 'grpc' },
   ],
+  buildService: config => {
+    if (config.type === 'grpc') {
+      return new GrpcGraphQLDataSource(config);
+    }
+    return new RemoteGraphQLDataSource(config);
```

Comment:
are we trying to handle both grpc and http case now?

Reply:
I left it in there during my experimentation. I don't see any harm in allowing both in case some service really wants to expose an HTTP /graphql endpoint, but we will encourage the use gRPC